src/main/java/picard/sam/AbstractAlignmentMerger.java

```diff
@@ -796,41 +796,57 @@ private void transferAlignmentInfoToPairedRead(final SAMRecord firstUnaligned, f
      * starts and ends of each read are the same.
      */
     protected static void clipForOverlappingReads(final SAMRecord read1, final SAMRecord read2, final boolean useHardClipping) {
-        // If both reads are mapped, see if we need to clip the ends due to small insert size
-        if (!(read1.getReadUnmappedFlag() || read2.getReadUnmappedFlag())) {
-            if (read1.getReadNegativeStrandFlag() != read2.getReadNegativeStrandFlag()) {
-                final SAMRecord pos = (read1.getReadNegativeStrandFlag()) ? read2 : read1;
-                final SAMRecord neg = (read1.getReadNegativeStrandFlag()) ? read1 : read2;
-
-                // Innies only -- do we need to do anything else about jumping libraries?
-                if (pos.getAlignmentStart() < neg.getAlignmentEnd()) {
-                    // first we softclip the 3' end of each read so that its 3' aligned end does not extends past the 5' aligned start of it's mate
-                    clip3primeEndsTo5primeEnds(pos, neg, false, false);
-
-                    if (useHardClipping) {
-                        // if we want to hardclip, we additionally hardclip the 3' end of each read so that its 3' unclipped end does not extend past the 5' unclipped start of its mate
-                        clip3primeEndsTo5primeEnds(pos, neg, true, true);
-                    }
-                }
+        // Only clip if both reads are mapped, on opposite strands, overlapping
+        if (!read1.getReadUnmappedFlag() && !read2.getReadUnmappedFlag() &&
+                read1.getReadNegativeStrandFlag() != read2.getReadNegativeStrandFlag() &&
+                read1.overlaps(read2)) {
```

Comment:
The explicit "innie" requirement is dropped, but this still will not clip outies, because I believe by definition outies cannot be overlapping with reads on opposite strands.  I don't think innie and outie really have meaning for overlapping reads, though I may be confused about this.  There is a test commented as `Non overlapping "outies"` which tests outies.

Previously, the "innie" requirement was enforced with the logic `pos.getAlignmentStart() < neg.getAlignmentEnd()`. Combining this with `posClipFrom > 0`/ `negClipFrom > 0` is _almost_ equivalent to the overlapping requirement, except for when the two reads overlap by only a single base.  In this case, the new check of `read1.overlaps(read2)` will pass, while the old check of `pos.getAlignmentStart() < neg.getAlignmentEnd()` will not, because instead `pos.getAlignmentStart() == neg.getAlignmentEnd()`.  This is not a behavior change I was intending, as I wasn't aware of this edge case difference until just now.  But I believe the old behavior was actually a bug, since the new behavior is more consistent with my understanding of what is intended.  It doesn't make sense to me that we would clip when two or more bases overlap, but not when only one base overlaps, so I'm guessing that was not intentional.  I have added a test demonstrating this edge case (commented as "overlap by one base only"), and have also added the same test on the branch [ck_mba_single_overlap_test_example](https://github.com/broadinstitute/picard/tree/ck_mba_single_overlap_test_example) to demonstrate the previous behavior.

For your softclipped example, these reads will not go through any additional clipping since they are not overlapping. 
 This is consistent with past behavior, since they would not pass the `pos.getAlignmentStart() < neg.getAlignmentEnd()` check.

Reply:
👍 thanks for the explanation